File: python/sync_with_qdrant/src/appwrite.py
import os
from appwrite.client import Client
from appwrite.models import Document
from appwrite.services.databases import Databases
from appwrite.query import Query

def get_all_documents(api_key) -> list[Document]:
    client = Client()
    client.set_endpoint(os.environ.get("APPWRITE_FUNCTION_API_ENDPOINT"))
    client.set_project(os.environ["APPWRITE_FUNCTION_PROJECT_ID"])
    client.set_key(api_key)

    databases = Databases(client)

    cursor = None
    cumulative: list[Document] = []

    while True:
        queries = [Query.limit(1000)]
        if cursor:
            queries.append(Query.cursor_after(cursor))

        response = databases.list_documents(
            os.environ["APPWRITE_DATABASE_ID"],
            os.environ["APPWRITE_COLLECTION_ID"],
            queries,
        )

        documents: list[Document] = response.documents

        if len(documents) > 0:
            cursor = documents[-1].id
        else:
            break
        cumulative.extend(documents)

    return cumulative

File: node/sync-with-qdrant/src/appwrite.js
import { Client, Databases, Query } from 'node-appwrite';

class AppwriteService {
  constructor(apiKey) {
    const client = new Client();
    client
      .setEndpoint(process.env.APPWRITE_FUNCTION_API_ENDPOINT)
      .setProject(process.env.APPWRITE_FUNCTION_PROJECT_ID)
      .setKey(apiKey);

    this.databases = new Databases(client);
  }

  /**
   * Retrieves the entire collection of documents from an Appwrite database.
   *
   * @param {string} databaseId
   * @param {string} collectionId
   * @returns {Promise<import('node-appwrite').Models.Document[]>}
   */
  async getAllDocuments(databaseId, collectionId) {
    const cumulative = [];

    let cursor = null;
    do {
      const queries = [Query.limit(100)];

      if (cursor) {
        queries.push(Query.cursorAfter(cursor));
      }

      const { documents } = await this.databases.listDocuments(
        databaseId,
        collectionId,
        queries
      );

      if (documents.length === 0) {
        break;
      }

      cursor = documents[documents.length - 1].$id;

      cumulative.push(...documents);
    } while (cursor);

    return cumulative;
  }
}

export default AppwriteService;
